ambry-store/src/main/java/com.github.ambry.store/BlobStoreCompactor.java

```diff
@@ -0,0 +1,842 @@
+    if (!recovering) {
+      commitIndexSegments(logSegmentNames);
+      if (srcIndex.hardDeleter != null) {
+        srcIndex.hardDeleter.resume();
+      }
+    }
+  }
+
+  /**
+   * Cleans up after the commit is complete. Cleaning up involves:
+   * 1. Dropping the log segments that are no longer relevant.
+   * 2. Deleting all associated index segment and bloom files.
+   * 3. Deleting the clean shutdown file associated with the index of the swap spaces.
+   * 4. Resetting state.
+   * @param recovering {@code true} if this function was called in the context of recovery. {@code false} otherwise.
+   * @throws IOException if there were I/O errors during cleanup.
+   */
+  private void cleanup(boolean recovering) throws IOException {
+    cleanupLogAndIndexSegments(recovering);
+    File cleanShutdownFile = new File(dataDir, TARGET_INDEX_CLEAN_SHUTDOWN_FILE_NAME);
+    if (cleanShutdownFile.exists() && !cleanShutdownFile.delete()) {
+      logger.warn("Could not delete the clean shutdown file {}", cleanShutdownFile);
+    }
+    resetStructures();
+  }
+
+  // copy() helpers
+
+  /**
+   * Sets up the state required for copying data by populating all the data structures, creating a log and index that
+   * wraps the swap spaces and pausing hard delete in the application log.
+   * @throws InterruptedException if hard delete could not be paused.
+   * @throws IOException if logs and indexes could not be set up.
+   * @throws StoreException if there were exceptions reading to writing to store components.
+   */
+  private void setupState() throws InterruptedException, IOException, StoreException {
+    CompactionDetails details = compactionLog.getCompactionDetails();
+    long highestGeneration = 0;
+    for (String segmentName : details.getLogSegmentsUnderCompaction()) {
+      highestGeneration = Math.max(highestGeneration, LogSegmentNameHelper.getGeneration(segmentName));
+    }
+    List<Pair<String, String>> targetSegmentNamesAndFilenames = new ArrayList<>();
+    List<LogSegment> existingTargetLogSegments = new ArrayList<>();
+    for (String segmentName : details.getLogSegmentsUnderCompaction()) {
+      long pos = LogSegmentNameHelper.getPosition(segmentName);
+      String targetSegmentName = LogSegmentNameHelper.getName(pos, highestGeneration + 1);
+      String targetSegmentFileName =
+          LogSegmentNameHelper.nameToFilename(targetSegmentName) + TEMP_LOG_SEGMENT_NAME_SUFFIX;
+      File targetSegmentFile = new File(dataDir, targetSegmentFileName);
+      if (targetSegmentFile.exists()) {
+        existingTargetLogSegments.add(new LogSegment(targetSegmentName, targetSegmentFile, metrics));
+      } else {
+        targetSegmentNamesAndFilenames.add(new Pair<>(targetSegmentName, targetSegmentFileName));
+      }
+    }
+    // TODO: available swap space count should be obtained from DiskManager. For now, assumed to be 1.
+    long targetLogTotalCapacity = srcLog.getSegmentCapacity();
```

Comment:
I don't think this is required since this will eventually be an API call to the segment pooler.